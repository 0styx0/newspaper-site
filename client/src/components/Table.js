import React from 'react';


/**
 * @prop headings - array
 * @prop rows - 2d array where [row][tdText]
 */
function Table(props) {

    return (<table>
        <thead>
            <tr>
                {props.headings.map((text, idx) => <th key={idx}>{text}</th>)}
            </tr>
        </thead>
        <tbody>
            {props.rows.map((row, idx) =>
                <tr key={idx}>
                    {row.map((cellText, idx) => <td key={idx}>{cellText}</td>)}
                </tr>)}
        </tbody>
    </table>);
}

export default Table;