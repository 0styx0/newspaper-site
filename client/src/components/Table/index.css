table,
td,
th {
    border-collapse: collapse;
    padding: 1vw;
    border: 1px solid #000;
    text-align: center;
    border-spacing: 0px;
}

table ~ label {
    text-align: left;
}

