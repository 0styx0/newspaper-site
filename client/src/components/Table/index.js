import React from 'react';

import './index.css';

/**
 * @prop headings - array
 * @prop rows - 2d array where [row][tdText]
 */
class Table extends React.Component {

    render() {

        return (<table>
            <thead>
                <tr>
                    {this.props.headings.map((text, idx) => <th key={idx}> {text && text[0] ? text[0].toUpperCase() + text.slice(1) : text}</th>)}
                </tr>
            </thead>
            <tbody>
                {this.props.rows.map((row, idx) =>
                    <tr key={idx}>
                        {row.map((cellText, idx) => <td key={idx}>{cellText}</td>)}
                    </tr>)}
            </tbody>
        </table>);
    }
}

export default Table;