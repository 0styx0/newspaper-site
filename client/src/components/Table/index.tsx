import * as React from 'react';

import './index.css';

interface Props {
    headings: Array<Element | JSX.Element | string>;
    rows: any[][];
}


/**
 * @prop headings - array
 * @prop rows - 2d array where [row][tdText]
 */
function Table(props: Props) {

    return (
        <table>
            <thead>
                <tr>
                    {props.headings.map((text, idx) => <th key={idx}> {text && text[0] ?
                        // the toString is purely to satisfy typescript
                        text[0].toUpperCase() + text.toString().slice(1) : text}</th>)}
                </tr>
            </thead>
            <tbody>
                {props.rows.map((row, idx) =>
                    <tr key={idx}>
                        {row.map((cellText: Text | Element, tdIdx: number) => <td key={tdIdx}>{cellText}</td>)}
                    </tr>)}
            </tbody>
        </table>
    );
}

export default Table;
