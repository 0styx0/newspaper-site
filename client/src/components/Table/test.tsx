import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Table from './';
import casual from '../../tests/casual.data';



describe('<Slideshow>', () => {

    describe('snapshots', () => {

        function snap(headings: any[], rows: any[][]) {

            const tree = renderer.create(
                <Table headings={headings} rows={rows} />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        }

        it('renders correctly', () => {

            const headings = ['col1', 'col2'];

            const rows = [
                ['hi', 'bye'],
                ['yes', 'no'],
                ['short', 'tall']
            ];

            snap(headings, rows);
        });

    });
});