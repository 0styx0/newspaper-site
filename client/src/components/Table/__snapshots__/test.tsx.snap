// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slideshow> snapshots renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <th>
         
        Col1
      </th>
      <th>
         
        Col2
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        hi
      </td>
      <td>
        bye
      </td>
    </tr>
    <tr>
      <td>
        yes
      </td>
      <td>
        no
      </td>
    </tr>
    <tr>
      <td>
        short
      </td>
      <td>
        tall
      </td>
    </tr>
  </tbody>
</table>
`;
